zarr/meta.py

```diff
@@ -10,11 +10,14 @@
 from zarr.errors import MetadataError
 
 
-def decode_metadata(b):
+ZARR_FORMAT = 2
+
+
+def decode_array_metadata(b):
     s = text_type(b, 'ascii')
     meta = json.loads(s)
     zarr_format = meta.get('zarr_format', None)
-    if zarr_format != 1:
+    if zarr_format != ZARR_FORMAT:
```

Comment:
I was thinking to add a migration function, but I was going to do that
later. I have some other work in a separate branch on support for filters
that will also go into v2 and I thought I'd deal with migration last.

On Thursday, August 25, 2016, Stephan Hoyer <notifications@github.com>
wrote:

> In zarr/meta.py
> <https://github.com/alimanfoo/zarr/pull/37#discussion_r76265993>:
>
> >      s = text_type(b, 'ascii')
> >      meta = json.loads(s)
> >      zarr_format = meta.get('zarr_format', None)
> > -    if zarr_format != 1:
> > +    if zarr_format != ZARR_FORMAT:
>
> I guess there are probably few users of v1? Might be worth considering
> making zarr backwards compatible (and/or releasing an upgrade script), at
> least for future version changes.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/alimanfoo/zarr/pull/37/files/35c250bc1bb8a6292a939eaef69eaf665717d07d#r76265993>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAq8QvwPhhFID-znyrB_H_lkKG9_98-Mks5qjbSygaJpZM4JUIhp>
> .
>


-- 
Alistair Miles
Head of Epidemiological Informatics
Centre for Genomics and Global Health <http://cggh.org>
The Wellcome Trust Centre for Human Genetics
Roosevelt Drive
Oxford
OX3 7BN
United Kingdom
Email: alimanfoo@googlemail.com
Web: http://purl.org/net/aliman
Twitter: https://twitter.com/alimanfoo
Tel: +44 (0)1865 287721